hamilton/driver.py

```diff
@@ -840,6 +854,290 @@ def visualize_path_between(
         except ImportError as e:
             logger.warning(f"Unable to import {e}", exc_info=True)
 
+    def materialize(
+        self,
+        *materializers: materialization.MaterializerFactory,
+        additional_vars: List[Union[str, Callable, Variable]],
+        overrides: Dict[str, Any] = None,
+        inputs: Dict[str, Any] = None,
+    ) -> Tuple[Any, Dict[str, Any]]:
+        """Executes and materializes with ad-hoc materializers.
+        This does the following:
+            1. Creates a new graph, appending the desired materialization nodes
+            2. Runs the graph with the materialization nodes outputted, as well as any additional
+                nodes requested (which can be empty)
+            3. Returns a Tuple[Materialization metadata, additional_vars result]
+
+        For instance, say you want to Materialize the output of a node to CSV:
+
+        .. code-block:: python
+             from hamilton import driver, base
+             from hamilton.io.materialize import to
+             dr = driver.Driver(my_module, {})
+             # foo, bar are pd.Series
+             metadata, result = dr.Materialize(
+                 to.csv(
+                     path="./output.csv"
+                     id="foo_bar_csv",
+                     dependencies=["foo", "bar"],
+                     combine=base.PandasDataFrameResult()
+                 ),
+                 additional_vars=["foo", "bar"]
+             )
```

Comment:
I think this whitespace is required.
```suggestion
        .. code-block:: python

             from hamilton import driver, base
             from hamilton.io.materialize import to
             dr = driver.Driver(my_module, {})
             # foo, bar are pd.Series
             metadata, result = dr.Materialize(
                 to.csv(
                     path="./output.csv"
                     id="foo_bar_csv",
                     dependencies=["foo", "bar"],
                     combine=base.PandasDataFrameResult()
                 ),
                 additional_vars=["foo", "bar"]
             )
```